src/test/java/com/recurly/v3/PagerTest.java

```diff
@@ -0,0 +1,83 @@
+package com.recurly.v3;
+
+import com.recurly.v3.fixtures.MockClient;
+import com.recurly.v3.fixtures.MyResource;
+import okhttp3.*;
+import okhttp3.Request;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+public class PagerTest {
+  @Test
+  public void testForEach() {
+    OkHttpClient mockOkHttpClient = getMockOkHttpClient(getResourceListJson());
+
+    final MockClient client = new MockClient("siteId", "apiKey", mockOkHttpClient);
+    Pager<MyResource> pager = client.listResources(null);
+    pager.forEach(resource -> assertNotNull(resource.getMyString()));
+  }
+
+  @Test
+  public void testEmptyList() {
+    OkHttpClient mockOkHttpClient = getMockOkHttpClient(getEmptyListJson());
+
+    final MockClient client = new MockClient("siteId", "apiKey", mockOkHttpClient);
+    Pager<MyResource> pager = client.listResources(null);
+    assertEquals(0, pager.getData().size());
+
+    for (MyResource myResource : pager) {
+      System.out.println("This should not throw exception" + myResource.getMyString());
+    }
+    pager.forEach(myResource -> System.out.println("This should not throw exception either" + myResource.getMyString()));
+  }
+
+  private OkHttpClient getMockOkHttpClient(String responseJson) {
+    final OkHttpClient mockOkHttpClient = mock(OkHttpClient.class);
+    final Call mCall = mock(Call.class);
+    final Request mRequest = new Request.Builder().url("https://api.recurly.com").build();
+
+    final Response mResponse =
```

Comment:
If we only have one page, how are we testing the pagination logic? If you use Mockito, you can setup a sequence of response. Does mockwebserver support this?

Reply:
MockWebServer supports multiple pages, but so does Mockito. I assume this comment was before the latest code was pushed, so I'm going to resolve this conversation. Feel free to comment on the latest code if you feel this concern still exists.